### PR TITLE
LG-12369 add accessibility checks to dashboard

### PR DESCRIPTION
**Pull request by Sgtpluck**
### Relevant Ticket or Conversation:
https://cm-jira.usa.gov/browse/LG-12369

### Description of Changes:

This PR:
* Updates our accessibility test library
* Adds accessibility Rspec matchers to have parity with the IdP
* Updates tests
* Updates inaccessible code

### PR Checklist:

1. [ ] Have you linted and tested your code locally prior to submission?
2. [ ] Have you tagged the appropriate dev(s) for review?
3. [ ] Have you linked to any relevant tickets or conversations?

### PR Review Standards: 
- Consider using [Conventional Comments](https://conventionalcomments.org/) to ensure that your feedback is clear and actionable.
- Ideally, PRs should be reviewed by at least 2 team members.
- All PRs must be approved before being merged.

### CODE REVIEW

File: spec/support/matchers/accessibility.rb [hunk 1]
```diff
@@ -0,0 +1,316 @@
+RSpec::Matchers.define :have_valid_idrefs do
```
Comment (zachmargolis) [inline spec/support/matchers/accessibility.rb:1]: just a question: is this copied from IDP?

**Review by zachmargolis — APPROVED** [review]

File: spec/support/matchers/accessibility.rb (hunk 1 above)
Comment (Sgtpluck) [reply spec/support/matchers/accessibility.rb:1]: yes! i mentioned it briefly in the PR comment, but meant to explicitly comment on the file to say that

File: app/views/service_providers/show.html.erb [hunk 2]
```diff
@@ -12,21 +12,21 @@
   <label for="name">App name: </label>
   <span class="font-mono-xs margin-left-5"><%= service_provider.app_name %></span>
 </h2>
-<p><i> <%= sanitize I18n.t('service_provider_form.app_name') %> </i></p>
+<p class="text-italic"><%= sanitize I18n.t('service_provider_form.app_name') %></p>
```
[the remark this answers is not in the document]
Comment (Sgtpluck) [reply app/views/service_providers/show.html.erb:15]: oh i can fix that right now!

[the remark this answers is not in the document]
Comment (Sgtpluck) [reply app/views/service_providers/show.html.erb]: updated in https://github.com/18F/identity-dashboard/pull/762/commits/49336340647d0b00218d72d8baf65a98680633db

File: app/views/service_providers/show.html.erb [hunk 3]
```diff
@@ -40,22 +40,22 @@
...
-<p><i> <%= I18n.t('service_provider_form.protocol') %> </i></p>
+<p class="text-italic"><%= I18n.t('service_provider_form.protocol') %></p>
 <hr>
 
 <h2 class="margin-bottom-0">
   <label for="ial_friendly">Level of Service:</label>
   <span class="font-mono-xs margin-left-5"><%= service_provider.ial_friendly %></span>
 </h2>
-<p><i> <%= I18n.t('service_provider_form.identity_assurance_level').html_safe %> </i></p>
+<div class="text-italic"><%= I18n.t('service_provider_form.identity_assurance_level').html_safe %></div>
```
[the remark this answers is not in the document]
Comment (Sgtpluck) [reply app/views/service_providers/show.html.erb]: updated in https://github.com/18F/identity-dashboard/pull/762/commits/49336340647d0b00218d72d8baf65a98680633db

File: app/views/service_providers/_form.html.erb [hunk 4]
```diff
@@ -344,64 +344,48 @@ B7xxt8BVc69rHV15A0qyx77CLSj3tCx2IUXVqRs5mlvA==
...
-          <%= f.input :es, label: "Español", as: :text, required: false, input_html: { value: service_provider.help_text.dig("forgot_password", "es") } %>
-          <%= f.input :fr, label: "Français", as: :text, required: false, input_html: { value: service_provider.help_text.dig("forgot_password", "fr") } %>
+          <%= f.input  :en, label: "English", as: :text, required: false, input_html: { value: service_provider.help_text.dig("forgot_password", "en"), readonly: readonly_help_text? } %>
+          <%= f.input :es, label: "Español", as: :text, required: false, input_html: { value: service_provider.help_text.dig("forgot_password", "es"), readonly: readonly_help_text? } %>
+          <%= f.input :fr, label: "Français", as: :text, required: false, input_html: { value: service_provider.help_text.dig("forgot_password", "fr"), readonly: readonly_help_text? } %>
         <% end %>
-    <% end %>
     </fieldset>
-  <% else %>
-    <% if @help_text_empty %>
```
Comment (Sgtpluck) [inline app/views/service_providers/_form.html.erb:380]: i removed this big duplicative conditional which had a lot of invalid html, and instead made the `readonly` a conditional

File: app/views/service_providers/show.html.erb [hunk 5]
```diff
@@ -170,39 +170,39 @@
...
-<p><i> <%= I18n.t('service_provider_form.attribute_bundle').html_safe %> </i></p>
+<p class="text-italic"><%= I18n.t('service_provider_form.attribute_bundle_html') %></p>
 <hr>
 
 <h2 class="margin-bottom-0">
   <label for="active">Accessible:</label>
   <span class="font-mono-xs"><%= service_provider.active? ? 'Yes' : 'No' %></span>
 </h2>
-<p><i> <%= I18n.t('service_provider_form.active').html_safe %> </i></p>
+<p class="text-italic"><%= I18n.t('service_provider_form.active') %></p>
```
Comment (nprimak) [inline app/views/service_providers/show.html.erb:205]: So many `<i>` 's 😬 . Thanks for cleaning this all up!

**Review by nprimak — APPROVED** [review]

**Pull request merged**